src/v/iceberg/partition.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/partition.h"
+
+namespace iceberg {
+
+struct transform_comparison_visitor {
+    template<typename T, typename U>
+    bool operator()(const T&, const U&) const {
+        return false;
+    }
```

Comment:
does this need a SFINAE or a static assert that the types aren't the same? seems like it would match here over everything below?